build.sbt

```diff
@@ -134,7 +137,7 @@ lazy val root = (project in file("."))
     Compile / packageSrc / publishArtifact := true)
 
 lazy val connect = (project in file("graphframes-connect"))
-  .dependsOn(root)
+  .dependsOn(root % "provided") // It will be provided in the final POM
```

Comment:
The problem is assembly cannot exclude "compile" / "runtime" scope dependencies from the resulted JAR. So, that is the workaround: using it as "provided" here and modify an output POM via `pomPostProcess`